### ISSUE TITLE
runtime panic: "index out of range" (runtime error: index out of range)

### ISSUE THREAD

**Issue by sonyafenge** [futurewei-cloud/arktos#552]
**What happened**:
when start kubemark hollow-nodes, found 1/500, or 2/5000 hollow-node pods failed to create with panic error. 

**What you expected to happen**:
all hollow-node create successfully.
**How to reproduce it (as minimally and precisely as possible)**:
run the command below to start kubemark and scale hollow-node:
```
$ export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=100GB KUBE_GCE_ZONE=us-east1-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-8 NUM_NODES=8 NODE_DISK_SIZE=200GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=node500 KUBE_GCE_NETWORK=node500 APISERVERS_EXTRA_NUM=0 WORKLOADCONTROLLER_EXTRA_NUM=0 ETCD_EXTRA_NUM=0 SHARE_PARTITIONSERVER=true ENABLE_KCM_LEADER_ELECT=false
$ ./cluster/kube-up.sh
$ ./test/kubemark/start-kubemark.sh
$ ./cluster/kubectl.sh scale replicationcontroller hollow-node -n kubemark --replicas=500
```
check all pods status and found
```
$ kubectl get pods -n kubemark --field-selector=status.phase!=Running
NAME                HASHKEY               READY   STATUS     RESTARTS   AGE
hollow-node-9cgd7   4877537797806849241   0/3     Init:0/1   0          3m25s
```
check kube-scheduler.log and kube-apiserver.log
```
$ cat ./kube-scheduler.log | grep hollow-node-9cgd7
I0808 02:07:52.323010       1 scheduling_queue.go:817] About to try and schedule pod system/kubemark/hollow-node-9cgd7
I0808 02:07:52.323076       1 scheduler.go:457] Attempting to schedule pod: system/kubemark/hollow-node-9cgd7
I0808 02:07:52.325392       1 scheduler_binder.go:258] AssumePodVolumes for pod "system/kubemark/hollow-node-9cgd7", node "node5000-2api-minion-group-3w06"
I0808 02:07:52.325454       1 scheduler_binder.go:268] AssumePodVolumes for pod "system/kubemark/hollow-node-9cgd7", node "node5000-2api-minion-group-3w06": all PVCs bound and nothing to do
I0808 02:07:52.325505       1 factory.go:733] Attempting to bind hollow-node-9cgd7 to node5000-2api-minion-group-3w06
I0808 02:07:52.327531       1 scheduler.go:594] pod system/kubemark/hollow-node-9cgd7 is bound successfully on node node5000-2api-minion-group-3w06, 51 nodes evaluated, 50 nodes were found feasible

$ cat ./kube-apiserver.log | grep hollow-node-9cgd7
I0808 02:07:52.327125       1 wrap.go:47] POST /api/v1/tenants/system/namespaces/kubemark/pods/hollow-node-9cgd7/binding: (1.25958ms) 201 [kube-scheduler/v2.7.4 (linux/amd64) kubernetes/1cea4a1/scheduler [::1]:45648]
I0808 02:07:52.327475       1 graph_populator.go:150] updatePod kubemark/hollow-node-9cgd7 for node node5000-2api-minion-group-3w06
I0808 02:07:55.622491       1 wrap.go:47] GET /api/v1/tenants/system/namespaces/kubemark/pods/hollow-node-9cgd7: (4.385559ms) 200 [kubelet/v2.7.4 (linux/amd64) kubernetes/1cea4a1 34.73.232.13:20483]
I0808 02:07:56.221125       1 wrap.go:47] PATCH /api/v1/tenants/system/namespaces/kubemark/pods/hollow-node-9cgd7/status: (3.000601ms) 200 [kubelet/v2.7.4 (linux/amd64) kubernetes/1cea4a1 34.73.232.13:20483]
I0808 02:22:25.023289       1 wrap.go:47] GET /api/v1/namespaces/kubemark/pods/hollow-node-9cgd7: (1.903058ms) 200 [kubectl/v2.7.4 (linux/amd64) kubernetes/4fa2fd3 35.193.21.135:57688]
I0808 02:22:25.061143       1 wrap.go:47] GET /api/v1/namespaces/kubemark/pods/hollow-node-9cgd7: (1.678767ms) 200 [kubectl/v2.7.4 (linux/amd64) kubernetes/4fa2fd3 35.193.21.135:57688]
I0808 02:22:25.099962       1 wrap.go:47] GET /api/v1/namespaces/kubemark/events?fieldSelector=involvedObject.name%3Dhollow-node-9cgd7%2CinvolvedObject.tenant%3Ddefault%2CinvolvedObject.namespace%3Dkubemark%2CinvolvedObject.uid%3Db4e48396-487b-4b99-8768-7bc8953ad0a1: (578.202µs) 400 [kubectl/v2.7.4 (linux/amd64) kubernetes/4fa2fd3 35.193.21.135:57688]
```

check kubelet log on vm which host failed pod, get :
```
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: E0808 02:07:54.545680    1525 remote_runtime.go:626] ListNetworkInterfaces on hollow-node-9cgd7- from runtime service failed: rpc error: code = Unknown desc = not implemented
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: I0808 02:07:54.554907    1525 kubelet.go:2043] SyncLoop (PLEG): "hollow-node-9cgd7_kubemark_system(b4e48396-487b-4b99-8768-7bc8953ad0a1)", event: &pleg.PodLifecycleEvent{ID:"b4e48396-487b-4b99-8768-7bc8953ad0a1", Type:"ContainerStarted", Data:"89e4a86c623ad32642c729239e4d7fb806c66ec3f72dc90b4f073d8af882a4f5"}
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: I0808 02:07:54.554963    1525 kubelet.go:2043] SyncLoop (PLEG): "hollow-node-9cgd7_kubemark_system(b4e48396-487b-4b99-8768-7bc8953ad0a1)", event: &pleg.PodLifecycleEvent{ID:"b4e48396-487b-4b99-8768-7bc8953ad0a1", Type:"ContainerDied", Data:"56be46cffa38d29af8df67f995f8735843cff9f8b596fb418daf117d44f420cd"}
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: E0808 02:07:54.555533    1525 runtime.go:73] Observed a panic: "index out of range" (runtime error: index out of range)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: goroutine 13615 [running]:
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x387c1a0, 0x7556d90)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:69 +0x7b
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51 +0x82
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: panic(0x387c1a0, 0x7556d90)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /usr/local/go/src/runtime/panic.go:522 +0x1b5
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodSpec).Workloads(0xc001d3d538, 0x41cd280, 0xc001d3d400, 0x0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/api/core/v1/types.go:3269 +0x5fb
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/apis/core/v1/helper/qos.GetPodQOS(0xc001d3d400, 0x0, 0x0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/apis/core/v1/helper/qos/qos.go:50 +0x17f
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).convertStatusToAPIStatus(0xc000d0a000, 0xc001d3d400, 0xc000a44f80, 0x0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_pods.go:1423 +0xa0
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).generateAPIPodStatus(0xc000d0a000, 0xc001d3d400, 0xc000a44f80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_pods.go:1330 +0xfa
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncPod(0xc000d0a000, 0x0, 0xc001d3d400, 0x0, 0xc000a44f80, 0x0, 0x0, 0xc002bf6900, 0xc0034bf2c0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1528 +0x4fb
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).managePodLoop.func1(0xc0014fbef8, 0xc000716230, 0xc001ddbee0, 0x0, 0x0)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:179 +0x269
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).managePodLoop(0xc000716230, 0xc002b1d560)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:189 +0x125
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod.func1(0xc000716230, 0xc002b1d560)
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:227 +0x62
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]: created by k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod
Aug 08 02:07:54 node5000-2api-minion-group-3w06 kubelet[1525]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:225 +0x3a3
```

**Anything else we need to know?**:

**Environment**:
- Arktos version (use `kubectl version`):
```
commit 1cea4a17478de9c3c43251f3b8836abb369d3c70
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

[Archive.zip](https://github.com/futurewei-cloud/arktos/files/5044551/Archive.zip)

**Comment by sonyafenge** [comment]
add failed pod description:
```
Name:           hollow-node-9cgd7
Namespace:      kubemark
Tenant:         system
Priority:       0
Node:           node5000-2api-minion-group-3w06/10.40.0.38
Start Time:     Sat, 08 Aug 2020 02:07:52 +0000
Labels:         name=hollow-node
Annotations:    <none>
Status:         Pending
IP:             
Controlled By:  ReplicationController/hollow-node
Init Containers:
  init-inotify-limit:
    Container ID:  
    Image:         busybox
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      sysctl
      -w
      fs.inotify.max_user_instances=1000
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-kk52j (ro)
Containers:
  hollow-kubelet:
    Container ID:  
    Image:         gcr.io/workload-controller-manager/kubemark:rz4t62
    Image ID:      
    Ports:         4194/TCP, 10250/TCP, 10255/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP
    Command:
      /bin/sh
      -c
      /kubemark --morph=kubelet --name=$(NODE_NAME)  --v=4 --v=4 --kubeconfig=/kubeconfig/kubelet.kubeconfig $(CONTENT_TYPE) --alsologtostderr 1>>/var/log/kubelet-$(NODE_NAME).log 2>&1
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:     40m
      memory:  100M
    Environment:
      CONTENT_TYPE:  <set to the key 'content.type' of config map 'node-configmap'>  Optional: false
      NODE_NAME:     hollow-node-9cgd7 (v1:metadata.name)
    Mounts:
      /kubeconfig from kubeconfig-volume (ro)
      /var/log from logs-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-kk52j (ro)
  hollow-proxy:
    Container ID:  
    Image:         gcr.io/workload-controller-manager/kubemark:rz4t62
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      /kubemark --morph=proxy --name=$(NODE_NAME)  --v=4 --v=4 --kubeconfig=/kubeconfig/kubeproxy.kubeconfig $(CONTENT_TYPE) --alsologtostderr 1>>/var/log/kubeproxy-$(NODE_NAME).log 2>&1
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:     20m
      memory:  102900Ki
    Environment:
      CONTENT_TYPE:  <set to the key 'content.type' of config map 'node-configmap'>  Optional: false
      NODE_NAME:     hollow-node-9cgd7 (v1:metadata.name)
    Mounts:
      /kubeconfig from kubeconfig-volume (ro)
      /var/log from logs-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-kk52j (ro)
  hollow-node-problem-detector:
    Container ID:  
    Image:         k8s.gcr.io/node-problem-detector:v0.4.1
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      /node-problem-detector --system-log-monitors=/config/kernel.monitor --apiserver-override="https://34.75.47.168:443?inClusterConfig=false&auth=/kubeconfig/npd.kubeconfig" --alsologtostderr 1>>/var/log/npd-$(NODE_NAME).log 2>&1
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:     20m
      memory:  20Mi
    Environment:
      NODE_NAME:  hollow-node-9cgd7 (v1:metadata.name)
    Mounts:
      /config from kernelmonitorconfig-volume (ro)
      /kubeconfig from kubeconfig-volume (ro)
      /var/log from logs-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from no-serviceaccount-access-to-real-master (ro)
Conditions:
  Type              Status
  Initialized       False 
  ContainersReady   False 
  Ready             False 
  PodScheduled      True 
Volumes:
  kubeconfig-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  kubeconfig
    Optional:    false
  kernelmonitorconfig-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      node-configmap
    Optional:  false
  logs-volume:
    Type:          HostPath (bare host directory volume)
    Path:          /var/log
    HostPathType:  
  no-serviceaccount-access-to-real-master:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  default-token-kk52j:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-kk52j
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 900s
Events:          <none>

```

**Comment by sonyafenge** [comment]
fixed by https://github.com/futurewei-cloud/arktos/pull/554.

**Comment by sonyafenge** [comment]
fixed by #554.
